services/apps/src/apps/clients/iplant_groups.clj

```diff
@@ -136,3 +138,25 @@
                     :form-params  (public-permission-update-body)
                     :content-type :json
                     :as           :json})))
+
+(defn- share-app*
+  "Shares an app with a user."
+  [app-id subject-id level]
+  (let [resource-name (grouper-app-resource-name app-id)
+        role-name     (grouper-user-group)]
+    (http/put (grouper-url "attributes" resource-name "permissions" "memberships" role-name subject-id level)
+              {:query-params {:user grouper-user}
+               :form-params  {:allowed true}
+               :content-type :json
+               :as           :json}))
+  nil)
+
+(defn share-app
+  "Shares an app with a user."
+  [app-id subject-id level]
+  (try+
+   (share-app* app-id subject-id level)
+   (catch :status {:keys [body]}
+     (let [reason (:grouper_result_message (service/parse-json body))]
```

Comment:
Should this perhaps have a default of some sort in case the error isn't directly from grouper -- e.g., if grouper-nginx sends something which errors? Getting "unable to share <uuid> with <subject>: " as an error wouldn't be as nice as something like "unable to share <uuid> with <subject>: non-grouper error caught"